### ISSUE TITLE
Rerunning E3SMv2 with maint-2.0 fails on chrysalis

### ISSUE THREAD

**Issue by golaz** [E3SM-Project/E3SM#5700]
I'm trying to rerun a portion of one the E3SMv2 simulation from the v2 simulation campaign to save additional output using current version of maint-2.0 on chrysalis (same machine as used originally).

Unfortunately, the model fails while initializing the land component. In particular, it fails during a check of PFT weights. Here is the error

 ```
Reading restart file v2.LR.amip_0101.elm.r.2000-01-01-00000.nc
 Reading restart dataset
 Opened existing file v2.LR.amip_0101.elm.r.2000-01-01-00000.nc         100
 ncd_inqvid: variable pfts1d_wttopounit is not on dataset
 check_weights ERROR: at t =           32 total PFT weight is 
  0.999999999992837      active_only =  T
 ENDRUN:
 ERROR in subgridWeightsMod.F90 at line 779
```
Files on chrysalis

/lcrc/group/e3sm/ac.golaz/E3SM/E3SMv2/v2.LR.amip_0101/tests/XS_1x10_ndays/run

Have there been changes to the ELM code that would cause this? Is there a new consistency check that needs to be disabled that was not present before?

**Comment by tangq** [comment]
@golaz , when I reran for the bonus output, [v2.LR.amip_bonus](https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3436576769/v2.LR.amip+bonus) has `check_finidat_fsurdat_consistency = .false.`. Is this flag false in your run?

**Comment by golaz** [comment]
@tangq : I already have a number of consistency checks disabled already:

```
&finidat_consistency_checks
 check_finidat_fsurdat_consistency = .false.
 check_finidat_year_consistency = .false.
/
&dynpft_consistency_checks
 check_dynpft_consistency = .false.
/
```
But maybe there is a new one we need to add to reproduce v2 simulations.

**Comment by golaz** [comment]
Tried adding

`check_finidat_pct_consistency = .false.`

but got the same failure.

**Comment by rljacob** [comment]
I looked at the maint-2.0 branch git log and its all machine config and SCORPIO/CIME updates except for changes to enable SSP370 and SSP585.

**Comment by rljacob** [comment]
Can you try on anvil?  That's were we test the maint-2.0 branch.

**Comment by golaz** [comment]
@rljacob : I haven't run anything on anvil in ages. I'm targeting chrysalis because I want to BFB reproduce previous simulations. Let me first try with the initial conditions from the beginning of the simulation rather than in the middle of the run.

**Comment by rljacob** [comment]
Can you tell me which hash of the code generated the restart you're trying to run from? IIRC, you started runs slightly before the 2.0 tag.

**Comment by golaz** [comment]
@rljacob : I've figured some out the logic and you were on the right track with your question.

The code base used for the original v2.LR.amip simulation (part of the official v2 Water Cycle simulation campaign) was 1e754ed78cdae71b8a82bfdc7a2ec6a89d3d94af (see https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3174924293/v2.LR.amip)

Here is the logic: trying to restart from the original elm restart file fails with the error above

/lcrc/group/e3sm/ac.golaz/E3SM/E3SMv2/v2.LR.amip_0101/original/archive/rest/2000-01-01-00000

Restarting from elm restart files from @tangq 's rerun of that simulation works

/lcrc/group/e3sm/ac.qtang/E3SMv2/v2.LR.amip_0101_bonus/archive/rest/2000-01-01-00000

The difference between the files is the addition of the new variable `pfts1d_wttopounit`

**Bottom line: maint-2.0 cannot reproduce/rerun the E3SMv2 WaterCycle simulations. This is a problem.**

There is still one thing I don't understand. I've also tried to rerun with the same code based as @tangq rerun 6275091262780c7a4b54ba56dcfa013ab1efa149 (https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3436576769/v2.LR.amip+bonus) and that failed as well. I don't understand that part.

**Comment by wlin7** [comment]
Apparently the wcprod tests failed to detect this type of issue due to changes in initial/restart files. A new prod test design is needed.

For example, the F20TR.eam-wcprod test uses 'startup' run-type, and has null finidat (i.e., no elm.r file is involved for starting the test). If the prod tests are designed to mirror as closely as possible the real production simulation, which is a hybrid run-type for F20TR simulation, this issue would have been caught.

**Comment by tangq** [comment]
> **Bottom line: maint-2.0 cannot reproduce/rerun the E3SMv2 WaterCycle simulations. This is a problem.**
> 
> There is still one thing I don't understand. I've also tried to rerun with the same code based as @tangq rerun [6275091](https://github.com/E3SM-Project/E3SM/commit/6275091262780c7a4b54ba56dcfa013ab1efa149) (https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3436576769/v2.LR.amip+bonus) and that failed as well. I don't understand that part.

Just a thought, is that possible the failure of reproducing my rerun is due to the machine changes after https://github.com/E3SM-Project/E3SM/commit/6275091262780c7a4b54ba56dcfa013ab1efa149?

**Comment by proteanplanet** [comment]
I am about to restart main-2.0 on Perlmutter using the V2 PI Control for a paper. Should I hold off?

**Comment by rljacob** [comment]
If you're grabbing a V2 PI control restart set, yes it likely won't work.

**Comment by rljacob** [comment]
@golaz to clarify:  which hash generated what you call "original elm restart file" ?

**Comment by rljacob** [comment]
I suspect that PR #4273 is responsible for this.   It introduced topounits into ELM and changed content of the elm restart files.

```
2.0.0 tag:                          1e754ed78cd,     Sept 29, 2021
PR #4273                               07ff719,          Oct 7, 2021
2.0.1 tag:                            5de0be04eed     Jan 14, 2022
Start of maint-2.0 branch      28390a3dc    March 8, 2022
2.0.2 tag                               6aa83b43           July 25, 2022
```
The fastest way to get things running would be to create a maint-2.0.0 branch before Oct 7, 2021 and update the machine files.

The better fix is to add code the elm to inquire if the topo unit variables are on the restart files and only read them if they are.   Unless pre-topo-unit elm restarts are just completely incompatible with post topo-unit elm restarts.  @bishtgautam or @tktesfa2014 ?

**Comment by bishtgautam** [comment]
@golaz I could not access the `*elm.r.*.nc`. Either the file has been moved or I don't have access to it.

```
cd /lcrc/group/e3sm/ac.golaz/E3SM/E3SMv2/v2.LR.amip_0101/tests/XS_1x10_ndays_00/run
ls -l ./v2.LR.amip_0101.elm.r.2000-01-01-00000.nc
lrwxrwxrwx 1 ac.golaz E3SM 117 May 18 10:42 ./v2.LR.amip_0101.elm.r.2000-01-01-00000.nc -> /lcrc/group/e3sm/ac.golaz/E3SM/E3SMv2/v2.LR.amip_0101/init/2000-01-01-00000/v2.LR.amip_0101.elm.r.2000-01-01-00000.nc
```

**Comment by bishtgautam** [comment]
@golaz Can you please point me to the run script that I should use to reproduce the error?

**Comment by golaz** [comment]
@rljacob : the original restart files that do not work are from the simulation that was part of the v2 simulation campaign: 

https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3174924293/v2.LR.amip

@tangq later reran that simulation to get extra output. Using thOse restart files works:

https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3436576769/v2.LR.amip+bonus

**Comment by golaz** [comment]
@bishtgautam : here is a script that should help you reproduce the problem.

[run.v2.LR.amip_0101_rerun-20230519-maint20.sh.txt](https://github.com/E3SM-Project/E3SM/files/11604202/run.v2.LR.amip_0101_rerun-20230519-maint20.sh.txt)

**Comment by bishtgautam** [comment]
@golaz I can reproduce the error. I'm now looking into if I can fix the error.

**Comment by bishtgautam** [comment]
PS: @tktesfa2014 is going to try another approach to fix this issue instead of reverting the topounit PR as done in PR-#5737

**Comment by tktesfa2014** [comment]
@bishtgautam and @rljacob
I am able to reproduce the error now. I will check the code and let you know if I am able to fix this issue.

**Comment by rljacob** [comment]
@tktesfa2014 is there a reason to not just go with the fix in #5737?   Unless there is some need to use topo units in simulations with the 2.0 code base, its fine to just remove it.
